static/js/compose_actions.js

```diff
@@ -72,32 +72,72 @@ export const _get_focus_area = get_focus_area;
 
 export function set_focus(msg_type, opts) {
     const focus_area = get_focus_area(msg_type, opts);
-    if (window.getSelection().toString() === "" || opts.trigger !== "message click") {
+    if (focus_area === "#compose-textarea" && opts.trigger === "toggle message") {
```

Comment:
Can you add a block comment at the top of this block explaining what we're trying to do in concept? 

I could also imagine making it a function `restore_focus_after_message_type_switch` ... or that this code is a workaround for not having a `e.preventDefault()` in the click handler (which would stop browser focus moving when doing this click in the first case), and we should use that alternative solution.

Reply:
I'm pretty sure this isn't a workaround for not having `e.preventDefault()` -- the cursor is supposed to be able to move when you switch. See [this thread](https://chat.zulip.org/#narrow/stream/6-frontend/topic/cursor.20position.20for.20Private.3C.3EStream/near/1374091) for more context. I can add that as a comment above for more clarity.

Reply:
`get_focus_area` handles most of that logic, but I felt like the part where we put the cursor specifically at the end of the compose box text has to happen here since it involves some different logic than `get_focus_area`